### ISSUE TITLE
Config setting for defaulting dnscrypt client to ues TCP

### ISSUE THREAD

**Issue by invis-z** [AdguardTeam/dnsproxy#266]
For some network settings, it would be handy if the default protocol for the DNSCrypt client (mayby also plain DNS) can be changed to TCP. 
https://github.com/AdguardTeam/dnsproxy/blob/master/upstream/upstream_dnscrypt.go#L85

**Comment by ameshkov** [comment]
@iJorgen DNSCrypt is very different from DoH, this must be an inaccuracy of the log.

**Comment by ameshkov** [comment]
I guess we could introduce something like `dnscrypt-tcp://` scheme in the upstream addresses for that.

**Comment by ameshkov** [comment]
Yeah, it'd be better if we could encode the information about preferred protocol somehow in the address itself.

@jedisct1 Frank, is there any space for extending DNS stamps spec with additional flags? "Prefer TCP/UDP" flag may also be useful for DoH addresses as it would allow specifying which HTTP version to use.

For example, currently we run two probe connections (TLS and QUIC) in order to choose which protocol to use. It would be nice if people could select the preferred one, encode it in the stamp and thus avoid "probing".

**Comment by ameshkov** [comment]
`props` seem to have plenty of space for different flags in it:
> props is a little-endian 64 bit value that represents informal properties about the resolver. It is a logical OR combination of the following values:

**Comment by jedisct1** [comment]
I'm not sure that this should be part of the stamp. It sounds better as an application config flag.

People willing to use TCP only are likely to want it globally, not for specific resolvers.

Duplicating all resolvers in the list to add a TCP only variant would also be a little awkward.

Finally, DNSCrypt was really designed to be used over UDP, not TCP.
